packages/react/presence/src/Presence.tsx

```diff
@@ -143,5 +143,24 @@ function getAnimationName(styles?: CSSStyleDeclaration) {
   return styles?.animationName || 'none';
 }
 
+// Before React 19 accessing `element.props.ref` will throw a warning and suggest using `element.ref`
+// After React 19 accessing `element.ref` does the opposite
+// https://github.com/facebook/react/pull/28348
+//
+// Access the ref using the method that doesn't yield a warning
+function getElementRef(element: React.ReactElement) {
+  // Pre React 19 there's a getter on `element.props.ref` that throws a warning when attempting to access it.
+  // This is safe to rely on. (As in... obviously, old React versions won't change).
+  // https://github.com/facebook/react/blob/408258268edb5acdfdbf77bc6e0b0dc6396c0e6f/packages/react/src/jsx/ReactJSXElement.js#L89-L99
+  const getter = Object.getOwnPropertyDescriptor(element.props, 'ref')?.get;
+  const hasPropWarning = getter && 'isReactWarning' in getter && getter.isReactWarning;
+
+  if (hasPropWarning) {
+    return (element as any).ref;
+  }
+
+  return element.props.ref;
+}
+
```

Comment:
Ref access instance 1 of 2

A bit messy checking the internals, but a naïve check like in https://github.com/radix-ui/primitives/pull/2811 will yield a warning pre React 19